python-package/xgboost/core.py

```diff
@@ -445,13 +481,9 @@ def __init__(self, data, label=None, missing=None,
         if isinstance(data, list):
             raise TypeError('Input data can not be a list.')
 
-        data, feature_names, feature_types = _maybe_pandas_data(data,
-                                                                feature_names,
-                                                                feature_types)
-
-        data, feature_names, feature_types = _maybe_dt_data(data,
-                                                            feature_names,
-                                                            feature_types)
+        data, feature_names, feature_types = _convert_dataframes(
+            data, feature_names, feature_types
+        )
 
         label = _maybe_pandas_label(label)
```

Comment:
We might need to implement the `cudf` equivalent for  below  too as when it does not raise a  exception when i provide it with a  label `dataframe` with multiple columns .

```
ValueError: DataFrame for label cannot have multiple columns
```

https://github.com/dmlc/xgboost/blob/4be4484f966096b3a8a3df277dbb07aa1bcce8d1/python-package/xgboost/core.py#L322-L338

Reply:
@VibhuJawa There will be an error in c++ level.  The c++ implementation should handle much as possible so that we don't have too much duplicated code across language bindings.